### PR TITLE
fix: mysql storage provider: No database selected

### LINKED ISSUE #2192: mysql storage provider: `No database selected`
Lesson learned from a downstream `Store` implementation: MySQL's `USE` statement is only good on the connection it was executed. This connection may be recycled over time (`sql.DB` manages a pool of connections), or the query may run on a different connection from the pool.

Reference: https://github.com/trustbloc/edge-core/issues/71

### PR DESCRIPTION
**Pull request by talwinder50**
closes #2192

Signed-off-by: talwinder.kaur <talwinder.kaur@securekey.com>

### CODE REVIEW

File: pkg/storage/store_test.go [hunk 1]
```diff
@@ -229,10 +228,3 @@ func verifyItr(t *testing.T, itr storage.StoreIterator, count int, prefix string
 	require.Empty(t, itr.Value())
 	require.Error(t, itr.Error())
 }
-
```
Comment (talwinder50) [inline pkg/storage/store_test.go:232]: failure while creating table db_prefix_key631219b5-a59f-44b3-94f0-52b52e21c6a6: Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '-a59f-44b3-94f0-52b52e21c6a6.db_prefix_key631219b5-a59f-44b3-94f0-52b52e21c6a6 . This uuid with new change of database.tableName it keeps failing with sql instance.

Comment (llorllale) [reply pkg/storage/store_test.go:232]: Quoting the table name with the backtick (`) should do the trick.

[Permitted characters in MySQL object identifiers](https://dev.mysql.com/doc/refman/8.0/en/identifiers.html):

> * Permitted characters in unquoted identifiers:
>    * ASCII: [0-9,a-z,A-Z$_] (basic Latin letters, digits 0-9, dollar, underscore)
>    * Extended: U+0080 .. U+FFFF 
> 
> * Permitted characters in quoted identifiers include the full Unicode Basic Multilingual Plane (BMP), except U+0000:
>   * ASCII: U+0001 .. U+007F
>   * Extended: U+0080 .. U+FFFF 
> * ASCII NUL (U+0000) and supplementary characters (U+10000 and higher) are not permitted in quoted or unquoted identifiers.
> * Identifiers may begin with a digit but unless quoted may not consist solely of digits.
> * Database, table, and column names cannot end with space characters. 
> 
> The identifier quote character is the backtick (`):

Comment (llorllale) [reply pkg/storage/store_test.go:232]: Please leave the randomized table names in the tests... it allows fast test re-runs without incurring the risk of test failures due to naming conflicts.

File: pkg/storage/mysql/mysqlstore.go [hunk 2]
```diff
@@ -192,18 +192,12 @@ func (s *sqlDBStore) Put(k string, v []byte) error {
...
 
-	// Use query is used to select the created database without this DDL operations are not permitted
-	_, err := s.db.Exec(fmt.Sprintf(useDBQuery, s.dbName))
-	if err != nil {
-		return fmt.Errorf("failed to use db %s: %w", s.dbName, err)
-	}
-
 	// create upsert query to insert the record, checking whether the key is already mapped to a value in the store.
-	createStmt := "INSERT INTO `" + s.tableName + "` VALUES (?, ?) ON DUPLICATE KEY UPDATE value=?"
+	insertStmt := "INSERT INTO " + s.tableName + " VALUES (?, ?) ON DUPLICATE KEY UPDATE value=?"
```
Comment (soluchok) [inline pkg/storage/mysql/mysqlstore.go:196]: Why did you remove ``` character?

Comment (talwinder50) [reply pkg/storage/mysql/mysqlstore.go]: failure while executing insert statement on table prefixdb_test.prefixdb_test: Error 1046: No database selected.
if i dont remove `character, dot doesnt get picked up so thats why i removed. but now hyphen in the uuid key string doesnt work without `

File: pkg/storage/store_test.go (hunk 1 above)
Comment (talwinder50) [reply pkg/storage/store_test.go]: updated

File: pkg/storage/mysql/mysqlstore.go (hunk 2 above)
Comment (llorllale) [reply pkg/storage/mysql/mysqlstore.go:191]: @soluchok backticks are already there in `s.tableName`: https://github.com/hyperledger/aries-framework-go/pull/2194/files#diff-5c03fdd2a42c9a160e4a9b056cb9db20R131

**Review by llorllale — APPROVED** [review]

**Review by soluchok — APPROVED** [review]

**Pull request merged**